### DISCUSSION TITLE
Cannot remove/update NTP tab issue

### DISCUSSION THREAD

**Discussion by DPetkovic-dev** [microsoft/MicrosoftEdge-Extensions#217] [General]
So as soon as a new tab is opened in the browser, I've got the logic to close it instantly using Chrome API's tabs.remove feature.
While it works fine on other browsers that are built on Chromium engine and tabs get closed immediately, Edge uses to throw an error saying "Cannot remove NTP tab."

The issue would be solved by making a redirection before closing the tab (Chrome API's tabs.update), which would remove its status of "NTP tab". However, on some devices then I encountered the same type of error "Cannot update NTP tab." and it seems like any action on a new empty tab cannot be made on Edge browser there.

Waiting for the tab to be fully loaded (complete status) before trying to close/update it again hasn't solved an issue.
![image](https://github.com/user-attachments/assets/5cf509ab-8606-4a72-a6eb-e5abfd804581)
Image provided displays the mentioned error message.

**Comment by ManikanthMSFT** [comment]
Hi @DPetkovic-dev, thank you for reporting this. We are looking into this and will let you know as soon as we have an update.

**Reply by ManikanthMSFT** [reply]
Hi @DPetkovic-dev, could you please help us to understand the scenario you are trying to accomplish. What is the scenario for which your extension is trying to close a tab. 

Also, can you share the sample extension for us to repro.

**Reply by DPetkovic-dev** [reply]
Hello @ManikanthMSFT.
Here's a [demo extension](https://github.com/DPetkovic-dev/demo-extension) where you can test the issue i'm talking about.

The scenario i want to close tab is actually as soon as it's opened, so imagine it like "prevent opening new tabs" feature.
I've left some comments in the background.js script so i hope it'll make it easier and faster for You to reproduce it.
Feel free to ask any additional questions if there's a need.

Regards.

**Reply by ManikanthMSFT** [reply]
Hi @DPetkovic-dev, thank you for writing back to us. We have forwarded this to our team they will review and once we get an update we will write back to you.

**Reply by ManikanthMSFT** [reply]
Hi @DPetkovic-dev, I wanted to provide you with an update regarding the issue you have been experiencing with the Edge browser extension. As you mentioned, the logic to close a new tab instantly using Chrome API's tabs.remove feature works fine on other browsers built on the Chromium engine. However, Edge throws an error saying "Cannot remove NTP tab."

Our team has investigated this issue and found that Edge does not permit extensions to close users' New Tab Page (NTP) if they do not have the permissions to modify NTP. To resolve this error, please ensure that the extension has declared permissions to modify NTP in the manifest.

We understand that you have been waiting for quite some time for a resolution, and we appreciate your patience. If you have any further questions or need additional assistance, please do not hesitate to reach out.

Thank you for your understanding.

**Reply by biplab-sutradhar** [reply]
What permission is required in the manifest.json file to immediately close a new tab using the Chrome API's tabs.remove?
